swagger_parser/lib/src/generator/fill_controller.dart

```diff
@@ -9,25 +9,45 @@ import 'models/universal_rest_client.dart';
 final class FillController {
   /// Constructor that accepts configuration parameters for creating files
   const FillController({
+    required OpenApiInfo openApiInfo,
+    required ProgrammingLanguage programmingLanguage,
+    required String clientPostfix,
+    required bool squishClients,
+    required bool freezed,
+    required bool enumsToJson,
+    required bool markFilesAsGenerated,
+  })  : _openApiInfo = openApiInfo,
+        _clientPostfix = clientPostfix,
+        _programmingLanguage = programmingLanguage,
+        _squishClients = squishClients,
+        _freezed = freezed,
+        _enumsToJson = enumsToJson,
+        _markFilesAsGenerated = markFilesAsGenerated;
+
+  /// Constructor for tests that accepts configuration parameters with default values to create files
+  const FillController.forTests({
```

Comment:
Please remove this constructor, there is no need to add a constructor with all required fields

Reply:
In this case, a constructor with mandatory fields is needed both to ensure that all parameters are passed, and to set all default values of configuration parameters in the Generator constructor so that only one axis of changes remains when default values are changed.

A separate constructor for tests is needed in order not to pass all parameters each time, but to set them as defaults only for testing purposes. For example, it makes no sense to set the `mark_files_as_generated` flag as true in tests.

Perhaps it should be renamed, but the current name seems to me to explain the essence of its use.